File: backend/shoppro-infra/src/main/java/com/shoppro/service/OperationService.java
package com.shoppro.service;

import com.shoppro.entity.AutomationRule;
import com.shoppro.entity.SopAudit;
import com.shoppro.entity.SopTemplate;
import com.shoppro.entity.WorkTask;
import java.util.List;

/**
 * <p>
 * Operation Service Interface
 * </p>
 *
 * @author shoppro
 * @since 2023-12-22
 */
public interface OperationService {

    // Automation Rules
    List<AutomationRule> listAutomationRules(Long tenantId, String triggerEvent, Boolean isActive);

    AutomationRule createAutomationRule(AutomationRule rule);

    AutomationRule updateAutomationRule(Long id, AutomationRule rule);

    void deleteAutomationRule(Long id);

    AutomationRule toggleAutomationRule(Long id);

    // SOP Templates
    List<SopTemplate> listSopTemplates(Long tenantId);

    SopTemplate createSopTemplate(SopTemplate sopTemplate);

    SopTemplate updateSopTemplate(Long id, SopTemplate sopTemplate);

    void deleteSopTemplate(Long id);

    // Work Tasks
    List<WorkTask> listWorkTasks(Long userId, String status);

    WorkTask createWorkTask(WorkTask workTask);

    void completeWorkTask(Long id);

    void applySopToCustomer(Long customerId, Long sopTemplateId); // Generate tasks from SOP

    // Sop Audits
    List<SopAudit> listSopAudits(Long tenantId, String status, String ruleName, int page, int size);

    long countSopAudits(Long tenantId, String status, String ruleName);

    String exportSopAudit(Long tenantId);
}

File: shoppro-app/src/services/operationService.ts
import { http } from './http';

export interface SopTemplate {
    id: number;
    name: string;
    description?: string;
    steps: string; // JSON string
    createdAt?: string;
}

export interface WorkTask {
    id: number;
    userId: number;
    customerId?: number;
    type: 'sop' | 'manual' | 'ai_suggestion';
    title: string;
    description?: string;
    status: 'pending' | 'completed' | 'cancelled';
    dueTime?: string;
    createdAt?: string;
}

export const operationService = {
    // SOP Templates
    listSopTemplates: () => {
        return http.get<SopTemplate[]>('/operation/sop');
    },
    createSopTemplate: (data: Partial<SopTemplate>) => {
        return http.post<SopTemplate>('/operation/sop', data);
    },
    deleteSopTemplate: (id: number) => {
        return http.delete(`/operation/sop/${id}`);
    },
    applySopToCustomer: (customerId: number, sopTemplateId: number) => {
        return http.post('/operation/sop/apply', null, { params: { customerId, sopTemplateId } });
    },

    // Work Tasks
    listWorkTasks: (userId: number, status?: string) => {
        return http.get<WorkTask[]>('/operation/tasks', { params: { userId, status } });
    },
    createWorkTask: (data: Partial<WorkTask>) => {
        return http.post<WorkTask>('/operation/tasks', data);
    },
    completeWorkTask: (id: number) => {
        return http.put(`/operation/tasks/${id}/complete`);
    }
};

export default operationService;
